src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -5,6 +5,7 @@
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Linq;
```

Comment:
a side note: if stable sort is not strictly necessary, we can get rid of Linq dependency from this code (in a separate PR) by replacing the usage of `.OrderBy()` in `ExpirePriorityBucket` either as:

```diff
-- stable sort with linq
++ unstable sort without linq

- foreach (CacheEntry entry in priorityEntries.OrderBy(entry => entry.LastAccessed))
+ priorityEntries.Sort((x, y) => DateTimeOffset.Compare(x.LastAccessed, y.LastAccessed));
+ foreach (CacheEntry entry in priorityEntries)
```

or if preserving the order in original list is essential:

```diff
- foreach (CacheEntry entry in priorityEntries.OrderBy(entry => entry.LastAccessed))
+ var sortedPriorityEntries = new List<CacheEntry>(priorityEntries);
+ sortedPriorityEntries.Sort((x, y) => DateTimeOffset.Compare(x.LastAccessed, y.LastAccessed));
+ foreach (CacheEntry entry in sortedPriorityEntries)
```

Reply:
> we can get rid of Linq dependency

this code has not popped out in my profiles, so I've not tried that. But I am sure that this would be a time vs memory tradeoff. @am11 please feel free to experiment with that and send a PR if you find the gains worth it.